projects/openapi-utilities/src/openapi/openapi2/openapi-2-traverser.ts

```diff
@@ -0,0 +1,891 @@
+                nextJsonPath
+              )}`
+            );
+          }
+        }
+        break;
+      case 'array':
+        const arrayItems = schema.items;
+        const nextJsonPath = jsonPointer.append(jsonPath, 'items');
+        if (!isObject(arrayItems)) {
+          console.warn(
+            `Expected an object at: ${getReadableLocation(
+              nextJsonPath
+            )}, found ${arrayItems}`
+          );
+        } else if (arrayItems && isNotReferenceObject(arrayItems)) {
+          yield* this.traverseSchema(
+            arrayItems,
+            nextJsonPath,
+            [...conceptualPath, 'items'],
+            {
+              ...location,
+              jsonSchemaTrail: [...(location.jsonSchemaTrail || []), 'items'],
+            }
+          );
+        } else {
+          console.warn(
+            `Expected a flattened spec, found a reference at: ${getReadableLocation(
+              nextJsonPath
+            )}`
+          );
+        }
+        break;
+      case 'string':
+      case 'number':
+      case 'integer':
+        break;
+    }
+  }
+
+  *traverseComponentSchema(
+    schema: OpenAPIV2.SchemaObject,
+    schemaName: string
+  ): IterableIterator<IFact> {
+    const jsonPath = jsonPointer.append(
+      '',
+      'components',
+      'schemas',
+      schemaName
+    );
+    const conceptualPath = jsonPointer.decode(jsonPath);
+    const conceptualLocation = { inComponentSchema: { schemaName } };
+    const nextJsonPath = jsonPointer.append(jsonPath, 'example');
+
+    if (!isObject(schema)) {
+      console.warn(
+        `Expected an object at: ${getReadableLocation(
+          nextJsonPath
+        )}, found ${schema}`
+      );
+    } else if (isNotReferenceObject(schema)) {
+      // @ limitation not supporting examples
+      // this.checkJsonTrail(jsonPath, schema);
+      // if (schema.example) {
+      //   yield this.onComponentSchemaExample(
+      //     schema.example,
+      //     nextJsonPath,
+      //     [...conceptualPath, 'example'],
+      //     conceptualLocation
+      //   );
+      // }
+    } else {
+      console.warn(
+        `Expected a flattened spec, found a reference at: ${getReadableLocation(
+          nextJsonPath
+        )}`
+      );
+    }
+  }
+
+  ///////////////////////////////////////////////////////////////////////////////////
+
+  *onSpecification(
+    specification: OpenAPIV2.Document
+  ): IterableIterator<FactVariant<OpenApiKind.Specification>> {
+    yield {
+      location: {
+        jsonPath: '',
+        conceptualPath: [],
+        conceptualLocation: {},
+        kind: OpenApiKind.Specification,
+      },
+      value: this.getSpecificationFact(specification),
+    };
+  }
+
+  getSpecificationFact(
+    specification: OpenAPIV2.Document
+  ): OpenApiSpecificationFact {
+    const { paths, schemes, security, ...specificationFact } = specification;
+    return { ...specificationFact, openapi: this.input?.swagger || '2.0' };
```

Comment:
weird and hacky...openapi version is called `swagger` in swagger. not sure what the right path is here, but it's not super material